### ISSUE TITLE
Any performance comparison between w/o mlp layer?

### ISSUE THREAD

**Issue by Yogurt928** [LAION-AI/CLAP#100]
Dear authors,

I saw that in your latest paper, there are mlp layers added after both audio encoder and text encoder.
While the default setting in code is without the mlp setting.
Is there any comparion on the performance of model w/o mlp layer?
Which one should be better?

Thanks.

**Comment by Yogurt928** [comment]
One more question, when use the mlp_loss, the logits are calculated by:

a_logits_per_audio = logit_scale_a * audio_features @ text_features_mlp.T
a_logits_per_text = logit_scale_a * text_features_mlp @ audio_features.T
t_logits_per_audio = logit_scale_t * audio_features_mlp @ text_features.T
t_logits_per_text = logit_scale_t * text_features @ audio_features_mlp.T

which means the mapping is:
audio_features & text_features_mlp
audio_features_mlp & text_features

why not calculate:
audio_features & text_features
audio_features_mlp & text_features_mlp ?

Thanks!

**Comment by RetroCirce** [comment]
Hi, thank you for your question! This is a good finding and let me explain this, which is a very important part in our paper.
The reason we put so many codes on mlp loss is referred from this paper: https://github.com/nerdyrodent/VQGAN-CLIP. It claims that the MLP loss by crossing the audio embedding and text embedding with another MLP layers and calculating a cross merging loss is better than the simple mlploss. 

However, in our case, we found this actually gets worse performance. Therefore, we **gave up** using the mlploss. The 'MLP layer' you see in the paper is not used for mlp loss, it is a normal mlp layer after audio encoder and text encoder. It is call [text_projection and audio_projection](https://github.com/LAION-AI/CLAP/blob/main/src/laion_clap/clap_module/model.py#L490-L494), which is different from [text_transform and audio_transforrm](https://github.com/LAION-AI/CLAP/blob/main/src/laion_clap/clap_module/model.py#L487-L489).

In the final implementation, we did not use the text_transform and audio_transform layers for mlp loss, we just used the simple clip loss with audio_projection and text_projection layer. As you can see in this [training script](https://github.com/LAION-AI/CLAP/blob/main/experiment_scripts/train-htsat-roberta.sh#L37-L66), we did not pass the clap_mlp_loss = True. 

So, the mlp layer we mentioned in the paper is audio_projection and text_projection for simple clip loss, not for mlp loss.

**Comment by Yogurt928** [comment]
I see. Thanks for your reply~
